### PR TITLE
Add endpoint for getting pipeline's secrets

### PR DESCRIPTION
**Pull request by d2lam**
Feature #148 
Fixes https://github.com/screwdriver-cd/screwdriver/issues/161

### CODE REVIEW

File: plugins/pipelines/listSecrets.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                        return reply([]);
+                    }
+
+                    return canAccess(credentials, secrets[0])
+                        .then(() => reply(secrets.map(s => {
+                            const output = s.toJson();
+
+                            delete output.value;
+
+                            console.log(output);
```
Comment (stjohnjohnson) [inline plugins/pipelines/listSecrets.js]: marp

**Comment by petey** [comment]
👍 

just_another

File: test/plugins/pipelines.test.js [hunk 2]
```diff
@@ -230,6 +253,82 @@ describe('pipeline plugin test', () => {
...
+                done();
+            });
+        });
+
+        it('returns 403 when the user does not have push permissions', (done) => {
+            userMock.getPermissions.withArgs(scmUrl).resolves({ push: false });
+
+            server.inject(options, (reply) => {
+                assert.equal(reply.statusCode, 403);
+                done();
```
Comment (stjohnjohnson) [inline test/plugins/pipelines.test.js]: It's fine for now, but `server.inject` returns a promise.  So all these tests can be written like:

```
return server.inject(options).then((reply) => {
...
```

**Comment by stjohnjohnson** [comment]
👍 

thirsty cat

**Pull request merged**